### PR TITLE
Added cgroup v2 memory implementation

### LINKED ISSUE #104: cgroup2: check list for v2 controller implementations
## Urgent (required to get CRI and Moby work)
- [x] pids: #103
- [x] cpu: #107
- [x] memory (`memory.stats`): #108 @Zyqsempai
- [x] `memory.current`, `memory.swap.current`, and other memory stat files  https://www.kernel.org/doc/Documentation/cgroup-v2.txt : #117 
- [x] OOM event FD: #115
- [x] io: #112
- [x] devices (BPF): #116
- [x] `ToggleControllers()` (issue: https://github.com/containerd/cgroups/issues/121): #124

Metrics functions are urgent.
`Update()` functions are not necessary for CRI and Moby.

## Non-urgent
- [x] `NewSystemd()` and `LoadSystemd()`: #125 
- [x] freezer: #103
- [ ] perf_event
- [x] cpuset: #111
- [x] rdma: #119 
- [x] [hugetlb](https://www.spinics.net/lists/cgroups/msg24003.html) (to appear in kernel 5.XX)

### PR DESCRIPTION
**Pull request by Zyqsempai**
for #104 
Added cgroup v2 memory implementation

Signed-off-by: bpopovschi <zyqsempai@mail.ru>

### CODE REVIEW

**Comment by Zyqsempai** [comment]
@AkihiroSuda @crosbymichael PTAL

**Comment by crosbymichael** [comment]
LGTM

File: v2/memory.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+func NewMemory(unifiedMountpoint string) (*memoryController, error) {
+	mc := &memoryController{
+		unifiedMountpoint: unifiedMountpoint,
+	}
+	ok, err := mc.Available("/")
+	if err != nil {
+		return nil, err
+	}
+	if !ok {
+		return nil, ErrCPUNotSupported
```
[the remark this answers is not in the document]
Comment (Zyqsempai) [reply v2/memory.go]: Oh, yep, sorry for that;)

**Comment by Zyqsempai** [comment]
@estesp Fixed.

**Pull request merged**